src/authenticator.rs

```diff
@@ -799,6 +800,61 @@ where
         Ok(())
     }
 
+    /// Rename credential
+    ///
+    /// Realized by loading FlatCredential object, changing its label,
+    /// writing under the new name and removing the previous file.
+    ///
+    /// Requires button confirmation before starting.
+    ///
+    /// returns: no data, except for the result code
+    /// Errors:
+    ///     - OperationBlocked if the new name is occupied already (checked by name hash)
+    ///     - NotFound, if the current credential can't be open (e.g. due to key not available) or deserialized
+    ///     - UnspecifiedNonpersistentExecutionError, if the old file cannot be removed, or serialization error
+    ///     - NotEnoughMemory, if new file cannot be written
+    ///     - SecurityStatusNotSatisfied, if the encryption key cannot be fetched
+    fn rename_credential<const R: usize>(
+        &mut self,
+        rename_req: command::RenameCredential<'_>,
+        _reply: &mut Data<R>,
+    ) -> Result {
+        // DESIGN Get operation confirmation from user before proceeding
+        self.user_present()?;
+
+        // DESIGN check if the target name is occupied already
+        self.err_if_credential_with_label_exists(rename_req.new_label)?;
+        if !self.credential_with_label_exists(rename_req.label) {
+            return Err(Status::NotFound);
+        }
+
+        let credential = {
+            let mut c = self
+                .load_credential(rename_req.label)
+                .ok_or(Status::NotFound)?;
+            c.label =
+                ShortData::from_slice(rename_req.new_label).map_err(|_| Status::NotEnoughMemory)?;
+            c
+        };
+
+        // Serialize the credential (implicitly) and store it
+        let filename = self.filename_for_label(rename_req.new_label);
+        self.state.try_write_file(
+            &mut self.trussed,
+            filename,
+            &credential,
+            credential.encryption_key_type,
+        )?;
+
+        // Remove old file name
+        let filename_old = self.filename_for_label(rename_req.label);
+        try_syscall!(self
+            .trussed
+            .remove_file(self.options.location, filename_old))
+        .map_err(|_| Status::UnspecifiedNonpersistentExecutionError)?;
```

Comment:
In theory this should be `UnspecifiedPersistentExecutionError` because data was written before that. But its not really important.

Reply:
I think what should be changed is the name of the error, the current name does not reflect the ISO7816 spec really well. But I don't think we want to start being nitpicky about 100% perfect spec compliance on that kind of thing it's wasted time (give n that we probably already don't in many places). It was more an FYI than a actual suggestion.

Reply:
I do not think about compliance here, but more about the expected cause, when this would be debugged in the future. We should follow the least astonishment principle anyways. I can change that to `UnspecifiedPersistentExecutionError` if this would add additional information to you (in this case, that the user data were changed).

Reply:
I don't think it would help *us* debugging. But some third party interfacing with it might, though anyway they will face inconsistent status codes in other places so not sure it would really help. Change it if you want. I don't think it's important either way. 

Reply:
No changes then